src/blur.js

```diff
@@ -1,94 +1,87 @@
-import {floor, max, min} from "./math.js";
-import {arrayify} from "./utils.js";
-
-function blurTransfer(V, r, n, vertical) {
-  if (!r) return; // radius 0 is a noop
+export function blur1(values, r) {
+  if (!((r = +r) >= 0)) throw new RangeError("invalid r");
+  if (!r) return values;
+  const temp = values.slice();
+  const width = Math.floor(values.length);
+  const height = 1;
+  blurh(values, temp, r, width, height);
+  blurh(temp, values, r, width, height);
+  blurh(values, temp, r, width, height);
+  return values;
+}
 
-  const iterations = 3, r0 = Math.floor(r);
-  // for a non-integer radius, interpolate between floor(r) and ceil(r)
-  if (r === r0) {
-    for (let i = 0; i < iterations; i++) {
-      blurTransferInt(V, r, n, vertical);
-    }
-  } else {
-    const frac = r - r0, frac_1 = 1 - frac;
-    const data = V[0].slice();
-    for (let i = 0; i < iterations; i++) {
-      blurTransferInt(V, r0 + 1, n, vertical);
-    }
-    const data_ceil = V[0];
-    V[0] = data;
-    if (r0 > 0) {
-      for (let i = 0; i < iterations; i++) {
-        blurTransferInt(V, r0, n, vertical);
-      }
-    }
-    for (let i = 0; i < data.length; i++) {
-      V[0][i] = V[0][i] * frac_1 + data_ceil[i] * frac;
-    }
+export function blur2(values, width, rx, ry = rx) {
+  if (!((rx = +rx) >= 0)) throw new RangeError("invalid rx");
+  if (!((ry = +ry) >= 0)) throw new RangeError("invalid ry");
+  if (!((width = Math.floor(width)) >= 0)) throw new RangeError("invalid width");
+  if (!width || (!rx && !ry)) return values;
+  const temp = values.slice();
+  const height = Math.floor(values.length / width);
+  if (rx && ry) {
+    blurh(temp, values, rx, width, height);
+    blurh(values, temp, rx, width, height);
+    blurh(temp, values, rx, width, height);
+    blurv(values, temp, ry, width, height);
+    blurv(temp, values, ry, width, height);
+    blurv(values, temp, ry, width, height);
```

Comment:
Note that the horizontal blur here is swapped since that it ends by copying into `temp`, such that the vertical blur can start by reading from `temp` and end by copying into `values`, resulting in a correct in-place blur.